### PR TITLE
Closes #6 Test Data: DM

### LINKED ISSUE #6: Test Data: DM
### Background Information

Add dm_neuro to the package, starting from code example for [dm_peds](https://github.com/pharmaverse/pharmaversesdtm/blob/main/data-raw/dm_peds.R). Keep 15 patients and modfiy birth dates and ages to make more realistic for alzheimer's.

Note: this test data will be moved to [{pharmaversesdtm}](https://github.com/pharmaverse/pharmaversesdtm/) after the first release of {admiralneuro}

### Definition of Done

 - [x] `data-raw` folder should contain new program `dm_neuro` that creates DM datasetDM test dataset added to {admiralneuro} package
 - [x] `R/data.R` should be updated to document the new DM dataset (see [here](https://github.com/pharmaverse/admiral/blob/main/R/data.R) for an example file)

### PR DESCRIPTION
**Pull request by cgpele**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
@cgpele looks really good, thanks! Only thing is some of our automated CICD checks are failing (things like documentation and code style). Please could you go through the PR checklist - that should give you ideas on how to fix them. Alternatively if you click on the failing check it will give you hints on why it's failing - see the lint check as an example:

image

Let me know how it goes - happy to work with you if unsure about any of them. Once done please re-request review.

**Comment by manciniedoardo** [comment]
Please ignore the NEWS.md item in our PR checklist as we are in our first release so don't need to compile it.

**Comment by manciniedoardo** [comment]
@cgpele one of the checks may be failing because we need to add `^data-raw$` as a new line to the `.Rbuildignore` file - please try that

File: data-raw/dm_neuro.R [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+library(admiral)
+
+
+# Read input test data from pharmaversesdtm ----
+dm <- pharmaversesdtm::dm
+
+# Convert blank to NA ----
+dm <- convert_blanks_to_na(dm)
+
+# Method to select 15
```
Comment (manciniedoardo) [inline data-raw/dm_neuro.R:15]: can you remove the commented code if unneeded please?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Could you also delete data-raw/xx.R as we don't need this template file anymore?

File: data-raw/dm_neuro.R (hunk 1 above)
Comment (cgpele) [reply data-raw/dm_neuro.R:15]: Yep will do, thanks!

**Comment by cgpele** [comment]
@manciniedoardo Do you agree to add "neuro" to inst/WORDLIST ?

**Review by manciniedoardo — APPROVED** [review]
LGTM!
